src/control/controlqml.cpp

```diff
@@ -0,0 +1,75 @@
+#include "control/controlqml.h"
+
+#include "moc_controlqml.cpp"
+
+ControlQml::ControlQml(QObject* parent)
+        : QObject(parent),
+          m_emptyString(""),
+          m_pControlProxy(nullptr) {
+}
+
+void ControlQml::setGroup(const QString& group) {
+    m_group = group;
+    initialize();
+}
+
+const QString& ControlQml::getGroup() const {
+    if (!m_pControlProxy) {
+        return m_emptyString;
+    }
+    return m_pControlProxy->getKey().group;
+}
+
+void ControlQml::setKey(const QString& key) {
+    m_key = key;
+    initialize();
+}
+
+const QString& ControlQml::getKey() const {
+    if (!m_pControlProxy) {
+        return m_emptyString;
+    }
+    return m_pControlProxy->getKey().item;
+}
+
+void ControlQml::setValue(double newValue) {
+    if (!m_pControlProxy) {
+        return;
+    }
+    m_pControlProxy->set(newValue);
+}
+
+double ControlQml::getValue() const {
+    if (!m_pControlProxy) {
+        return -1;
+    }
+    return m_pControlProxy->get();
+}
+
+void ControlQml::setParameter(double newValue) {
+    if (!m_pControlProxy) {
+        return;
+    }
+    m_pControlProxy->setParameter(newValue);
+}
+
+double ControlQml::getParameter() const {
+    if (!m_pControlProxy) {
+        return -1;
+    }
+    return m_pControlProxy->getParameter();
+}
+
+void ControlQml::initialize() {
+    if (!m_group.isEmpty() && !m_key.isEmpty()) {
+        m_pControlProxy = std::make_unique<ControlProxy>(
+                m_group, m_key, this, ControlFlag::AllowMissingOrInvalid);
+        m_pControlProxy->connectValueChanged(this, &ControlQml::controlProxyValueChanged);
+        m_pControlProxy->emitValueChanged();
+    }
+}
+
+void ControlQml::controlProxyValueChanged(double newValue) {
+    emit valueChanged(newValue);
+    emit parameterChanged(m_pControlProxy->getParameter());
```

Comment:
This is an odd hack... should ControlProxy or ControlDoublePrivate emit a parameterChanged signal?

Reply:
@daschuer any thoughts about this?

Reply:
Yes, you can connect here to ControlDoublePrivate directly.
I suggest to follow control/controlobjectscript.cpp pattern this way you can also get rid of extra null checks.